src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/BinaryPrimitives.ReverseEndianness.cs

```diff
@@ -0,0 +1,260 @@
+        /// <exception cref="ArgumentException">The <paramref name="destination"/>'s length is smaller than that of the <paramref name="source"/>.</exception>
+        public static void ReverseEndianness(ReadOnlySpan<long> source, Span<long> destination) =>
+            ReverseEndianness<long, Int64EndiannessReverser>(source, destination);
+
+        /// <summary>Copies every primitive value from <paramref name="source"/> to <paramref name="destination"/>, reversing each primitive by performing an endianness swap as part of writing each.</summary>
+        /// <param name="source">The source span to copy.</param>
+        /// <param name="destination">The destination to which the source elements should be copied.</param>
+        /// <remarks>The source and destination spans may overlap. The same span may be passed as both the source and the destination in order to reverse each element's endianness in place.</remarks>
+        /// <exception cref="ArgumentException">The <paramref name="destination"/>'s length is smaller than that of the <paramref name="source"/>.</exception>
+        [CLSCompliant(false)]
+        public static void ReverseEndianness(ReadOnlySpan<nuint> source, Span<nuint> destination) =>
+#if TARGET_64BIT
+            ReverseEndianness<long, Int64EndiannessReverser>(MemoryMarshal.Cast<nuint, long>(source), MemoryMarshal.Cast<nuint, long>(destination));
+#else
+            ReverseEndianness<int, Int32EndiannessReverser>(MemoryMarshal.Cast<nuint, int>(source), MemoryMarshal.Cast<nuint, int>(destination));
+#endif
+
+        /// <summary>Copies every primitive value from <paramref name="source"/> to <paramref name="destination"/>, reversing each primitive by performing an endianness swap as part of writing each.</summary>
+        /// <param name="source">The source span to copy.</param>
+        /// <param name="destination">The destination to which the source elements should be copied.</param>
+        /// <remarks>The source and destination spans may overlap. The same span may be passed as both the source and the destination in order to reverse each element's endianness in place.</remarks>
+        /// <exception cref="ArgumentException">The <paramref name="destination"/>'s length is smaller than that of the <paramref name="source"/>.</exception>
+        public static void ReverseEndianness(ReadOnlySpan<nint> source, Span<nint> destination) =>
+#if TARGET_64BIT
+            ReverseEndianness<long, Int64EndiannessReverser>(MemoryMarshal.Cast<nint, long>(source), MemoryMarshal.Cast<nint, long>(destination));
+#else
+            ReverseEndianness<int, Int32EndiannessReverser>(MemoryMarshal.Cast<nint, int>(source), MemoryMarshal.Cast<nint, int>(destination));
+#endif
+
+        private readonly struct Int16EndiannessReverser : IEndiannessReverser<short>
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
is this attribute really needed? it won't get inlined without it?

Reply:
It's necessary for the vector-based overloads.  I included it on this one for completeness, but I can remove it from the scalar ones.